tests/behat/filter.feature

```diff
@@ -0,0 +1,76 @@
+@mod @mod_studentquiz
+Feature: Filtering in Studentquiz view.
+
+  Background:
+    Given the following "courses" exist:
+      | fullname | shortname | category |
+      | Course 1 | C1        | 0        |
+
+    And the following "users" exist:
+      | username | firstname | lastname | email                |
+      | student1 | Sam1      | Student1 | student1@example.com |
+      | student2 | Sam2      | Student2 | student2@example.com |
+    And the following "course enrolments" exist:
+      | user     | course | role    |
+      | student1 | C1     | student |
+      | student2 | C1     | student |
+    And the following "activities" exist:
+      | activity    | name               | intro                     | course | idnumber     | publishnewquestion |
+      | studentquiz | StudentQuiz Test 1 | StudentQuiz description 1 | C1     | studentquiz1 | 1                  |
+    And the following "questions" exist:
+      | questioncategory               | qtype     | name            | questiontext          |
+      | Default for StudentQuiz Test 1 | truefalse | Test question 1 | Answer the question 1 |
+      | Default for StudentQuiz Test 1 | truefalse | Test question 2 | Answer the question 2 |
+
+  @javascript @_switch_window
+  Scenario: Check validation client in the advance filter.
+    When I am on the "StudentQuiz Test 1" "mod_studentquiz > View" page logged in as "admin"
+    And I should see "Show more..."
+    Then I click on "Show more..." "link"
```

Comment:
The 'Then' in a Behat test is meant ot indicate when some expected thing is being verified. That is defitely not what is happening here.

Possibly, the best place for it in this Scenario is line 75 '    And I should see "Test question 1"' or line 31 '    And I should see "You must enter a number here."'

Also, no benefit in checking 'I should see "Show more..."' just before you do 'I click on "Show more..." "link"'. The 'click' check will also test that it is visible, so this is redundant.